src/main/scala/it/pagopa/interop/agreementmanagement/model/persistence/AgreementPersistentBehavior.scala

```diff
@@ -185,12 +195,12 @@ object AgreementPersistentBehavior {
   def calcNewAgreementState(
     suspendedByProducer: Option[Boolean],
     suspendedByConsumer: Option[Boolean],
+    suspendedByPlatform: Option[Boolean],
     newState: PersistentAgreementState
-  ): PersistentAgreementState = (newState, suspendedByProducer, suspendedByConsumer) match {
-    case (Active, Some(true), _) => Suspended
-    case (Active, _, Some(true)) => Suspended
-    case _                       => newState
-  }
+  ): PersistentAgreementState =
+    (suspendedByProducer ++ suspendedByConsumer ++ suspendedByPlatform)
+      .collectFirst { case true if newState == Active => Suspended }
+      .getOrElse(newState)
```

Comment:
just tried a different approach, let me know your thoughts

Reply:
I have to admit that I find it a bit difficult to read, mainly because I don't know what the operator `++` does to a couple of `Option[Boolean]`. Let's keep it this way btw.

Reply:
the previous solution was verbose but more clear to read in my opinion, I don't have strong preferences here, I can revert